src/main/java/com/esri/samples/map/generate_offline_map_overrides/GenerateOfflineMapOverridesController.java

```diff
@@ -0,0 +1,282 @@
+    downloadArea.setSymbol(simpleLineSymbol);
+
+    // update the download area whenever the viewpoint changes
+    mapView.addViewpointChangedListener(viewpointChangedEvent -> updateDownloadArea());
+
+    // create a map with the portal item
+    map = new ArcGISMap(portalItem);
+    map.addDoneLoadingListener(() -> {
+      // enable the generate offline map button when the map is loaded
+      if (map.getLoadStatus() == LoadStatus.LOADED) {
+        generateOfflineMapButton.setDisable(false);
+      }
+      updateDownloadArea();
+    });
+
+    // set the map to the map view
+    mapView.setMap(map);
+  }
+
+  /**
+   * Called when the Generate offline map button is clicked. Builds parameters for the offline map task from the UI
+   * inputs and executes the task.
+   */
+  @FXML
+  private void generateOfflineMap() {
+    try {
+      // show the progress bar
+      progressBar.setVisible(true);
+
+      // create an offline map task with the map
+      OfflineMapTask offlineMapTask = new OfflineMapTask(map);
+
+      // get default offline map parameters for this task given the download area
+      ListenableFuture<GenerateOfflineMapParameters> generateOfflineMapParametersFuture = offlineMapTask
+          .createDefaultGenerateOfflineMapParametersAsync(downloadArea.getGeometry());
+
+      generateOfflineMapParametersFuture.addDoneListener(() -> {
+        try {
+          final GenerateOfflineMapParameters parameters = generateOfflineMapParametersFuture.get();
+
+          // get additional offline parameters (overrides) for this task
+          ListenableFuture<GenerateOfflineMapParameterOverrides> parameterOverridesFuture = offlineMapTask
+              .createGenerateOfflineMapParameterOverridesAsync(parameters);
+
+          parameterOverridesFuture.addDoneListener(() -> {
+            try {
+              GenerateOfflineMapParameterOverrides overrides = parameterOverridesFuture.get();
+
+              // get the export tile cache parameters for the base layer
+              OfflineMapParametersKey basemapParamKey = new OfflineMapParametersKey(
+                  mapView.getMap().getBasemap().getBaseLayers().get(0));
+              ExportTileCacheParameters exportTileCacheParameters =
+                  overrides.getExportTileCacheParameters().get(basemapParamKey);
+
+              // create a new sublist of LODs in the range requested by the user
+              exportTileCacheParameters.getLevelIDs().clear();
+              for (int i = minScaleLevelSpinner.getValue(); i < maxScaleLevelSpinner.getValue(); i++) {
+                exportTileCacheParameters.getLevelIDs().add(i);
+              }
+              // set the area of interest to the original download area plus a buffer
+              exportTileCacheParameters.setAreaOfInterest(GeometryEngine.buffer(downloadArea.getGeometry(),
+                  extentBufferDistanceSpinner.getValue()));
```

Comment:
When I set the spinner to max (500 as specified in the fxml file), the offline map created still matches exactly the geometry of the download area. I thought the buffer would give an extra 500m of basemap coverage/offline data outside of the download area: might be misunderstanding what this functionality is for, or could it be that it isn't handling as expected?

Reply:
@Rachael-E so JavaFX spinners are a little weird because by default, you have to hit Enter after typing in a value to commit it. Not sure if you were doing that. I will add a note to the readme to warn users.

Reply:
We could use a slider instead. But I felt like many users would want to directly input the value. What do you think?